### PR TITLE
Test enhancements.

### PR DESCRIPTION
**Pull request by skriebel**
I was troubleshooting a possible issue with the cidr filter and I created a few more test to verify that the filter was working as advertised.  It was.

I figured I would just send these upstream.

### CODE REVIEW

File: spec/filters/cidr_spec.rb [hunk 1]
```diff
@@ -35,6 +36,75 @@
...
+       filter {
+        cidr {
+          address => [ "%{clientip}"]
+          network => [ "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]
+          add_tag => [ "matched" ]
+        }
+      }
+    CONFIG
+
+    sample("clientip" => "192.168.0.30") do
```
Comment (purbon) [inline spec/filters/cidr_spec.rb]: could this test and the one after this merge into the same one? they use the same configuration.

File: spec/filters/cidr_spec.rb [hunk 2]
```diff
@@ -35,6 +36,75 @@
        insist { subject["tags"] }.nil?
     end
   end
+ 
+  # Test multple CIDR blocks passed into 'network'.  Make sure we try an
+  # IP in every range.
+
+  describe "IPV4 match, passing a list to network [192.168.0.30]" do
+   config <<-CONFIG
```
Comment (purbon) [inline spec/filters/cidr_spec.rb:44]: we might be able to move this config out, as is the same for all describe sections. Then do a sample for each test case. What do you think?

**Comment by purbon** [comment]
@human39 good work man enhancing test. Nowadays we're moving out of using this kind of test style in logstash, the direction is more close to the one you've seen in the patterns core project. I was wondering if you would like to refactor the test in this plugin to be more rspec friendly, what do you think?

If you like to do it, I can provide you with samples. 

Let me know.

File: spec/filters/cidr_spec.rb (hunk 2 above)
Comment (skriebel) [inline spec/filters/cidr_spec.rb:44]: Good point.  I'll make this change.

**Comment by skriebel** [comment]
@purbon Thank you. 

I made the change you recommended.

Unfortunately, I'm not sure if I can commit to re-factoring the tests.  However, feel free to send me any information you have of examples of the style of tests Elastic wants and I'll see if it is something I can do.

**Comment by jordansissel** [comment]
@purbon I'm OK deferring the test refactor.

Code LGTM. Tests pass.

**Pull request closed without merging**